### PR TITLE
Message log report

### PR DESCRIPTION
**Pull request by esoergel**
This is a handful of changes whose user-facing effects should be hidden by the SMS_LOG_CHANGES feature flag.  I'd definitely review commit-by-commit - I pulled out a large chunk of report code into a separate file, which makes the PR seem bigger than it is.

@nickpell @millerdev

### CODE REVIEW

File: corehq/apps/reports/standard/sms.py [hunk 1]
```diff
@@ -319,16 +349,22 @@ def fields(self):
...
     def headers(self):
-        header = DataTablesHeader(
+        headers = DataTablesHeader(*[header for header in [
             DataTablesColumn(_("Timestamp")),
             DataTablesColumn(_("User Name")),
             DataTablesColumn(_("Phone Number")),
             DataTablesColumn(_("Direction")),
             DataTablesColumn(_("Message")),
+            DataTablesColumn(_("Status")) if self.testing_changes else Ellipsis,
+            DataTablesColumn(_("Event"), sortable=False) if self.testing_changes else Ellipsis,
```
Comment (millerdev) [inline corehq/apps/reports/standard/sms.py]: Why `Ellipsis` instead of `None`?

**Review by millerdev — APPROVED** [review]
Nice! Only one comment.

File: corehq/apps/reports/standard/sms.py (hunk 1 above)
Comment (esoergel) [reply corehq/apps/reports/standard/sms.py]: I did return `None` initially (and used `filter(None, [])`), but then when I do this same pattern below, I was worried about other falsy types like the empty string being inadvertently excluded, so I figured `Ellipsis` would be more explicit.  Then I went back and changed this one for parity.

Comment (millerdev) [reply corehq/apps/reports/standard/sms.py]: `[x for x in ... if x is not None]` is better than `filter(None, ...)`

**Pull request merged**